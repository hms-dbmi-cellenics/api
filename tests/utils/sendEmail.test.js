const AWS = require('aws-sdk-mock');
const sendEmail = require('../../src/utils/sendEmail');
const { SUCCEEDED, FAILED } = require('../../src/api.v2/constants');
const { USER } = require('../test-utils/constants');
const buildPipelineStatusEmailBody = require('../../src/utils/emailTemplates/buildPipelineStatusEmailBody');

process.env.DOMAIN_NAME = 'localhost.test';

describe('sending email works', () => {
  const experimentId = 'other-id';
  let sendEmailSpy;
  beforeAll(() => {
    jest.resetModules();
    jest.clearAllMocks();
    sendEmailSpy = jest.fn(() => Promise.resolve({ data: { MessageId: '200' } }));
    AWS.mock('SES', 'sendEmail', sendEmailSpy);
  });
  beforeEach(() => {
    sendEmailSpy.mockReset();
  });

  it('sends success email', async () => {
    const message = { experiment: experimentId };
    const params = buildPipelineStatusEmailBody(message.experiment, SUCCEEDED, USER);
    sendEmail(params);
    expect(sendEmailSpy).toHaveBeenCalledTimes(1);
    expect(sendEmailSpy.mock.calls).toMatchSnapshot();
  });
  it('sends failure email', async () => {
    const message = { experiment: experimentId };
    const params = buildPipelineStatusEmailBody(message.experiment, FAILED, USER);
    sendEmail(params);
    expect(sendEmailSpy).toHaveBeenCalledTimes(1);
    expect(sendEmailSpy.mock.calls).toMatchSnapshot();
  });
});
