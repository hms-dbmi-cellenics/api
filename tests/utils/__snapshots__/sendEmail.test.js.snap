// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sending email works sends failure email 1`] = `
Array [
  Array [
    Object {
      "Destination": Object {
        "ToAddresses": Array [
          "johncena@gmail.com",
        ],
      },
      "Message": Object {
        "Body": Object {
          "Html": Object {
            "Charset": "UTF-8",
            "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello John,</h3>
            <p>Thanks for using Cellenics! <br/>
            <br />
              
      Unfortunately, when trying to run the analysis the data processing failed for your experiment: <br/>
      <a href=\\"https://test.com/experiments/other-id/data-processing\\">https://test.com/experiments/other-id/data-processing</a> <br/>
      We are working hard to identify and fix the issue. We will let you know as soon as your data is ready to explore. <br/>
      <br/><br/>
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
          },
          "Text": Object {
            "Charset": "UTF-8",
            "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello John,</h3>
            <p>Thanks for using Cellenics! <br/>
            <br />
              
      Unfortunately, when trying to run the analysis the data processing failed for your experiment: <br/>
      <a href=\\"https://test.com/experiments/other-id/data-processing\\">https://test.com/experiments/other-id/data-processing</a> <br/>
      We are working hard to identify and fix the issue. We will let you know as soon as your data is ready to explore. <br/>
      <br/><br/>
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
          },
        },
        "Subject": Object {
          "Charset": "UTF-8",
          "Data": "Cellenics experiment status",
        },
      },
      "Source": "some@email.com",
    },
    [Function],
  ],
]
`;

exports[`sending email works sends success email 1`] = `
Array [
  Array [
    Object {
      "Destination": Object {
        "ToAddresses": Array [
          "johncena@gmail.com",
        ],
      },
      "Message": Object {
        "Body": Object {
          "Html": Object {
            "Charset": "UTF-8",
            "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello John,</h3>
            <p>Thanks for using Cellenics! <br/>
            <br />
              
      The data processing pipeline has completed successfully and your data is now ready to explore:<br/>
      <a href=\\"https://test.com/experiments/other-id/data-processing\\">https://test.com/experiments/other-id/data-processing</a>
      <br />
      <br />
      Happy analysing! <br/><br/><br/>
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
          },
          "Text": Object {
            "Charset": "UTF-8",
            "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello John,</h3>
            <p>Thanks for using Cellenics! <br/>
            <br />
              
      The data processing pipeline has completed successfully and your data is now ready to explore:<br/>
      <a href=\\"https://test.com/experiments/other-id/data-processing\\">https://test.com/experiments/other-id/data-processing</a>
      <br />
      <br />
      Happy analysing! <br/><br/><br/>
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
          },
        },
        "Subject": Object {
          "Charset": "UTF-8",
          "Data": "Cellenics experiment status",
        },
      },
      "Source": "some@email.com",
    },
    [Function],
  ],
]
`;
